### PR TITLE
MEPTS-731 - Testing: Validate MQ CAT 13 in all DBs

### PR DESCRIPTION
**Pull request by iteliolucas**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java [hunk 1]
```diff
@@ -1705,7 +1705,7 @@ public CohortDefinition getMQC13P3DEN(String indicatorFlag) {
             hivMetadata.getYesConcept().getConceptId());
 
     CohortDefinition transferredIn =
-        qualityImprovement2020Queries.getTransferredInPatients(
+        QualityImprovement2020Queries.getTransferredInPatients(
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java:2145]: Yap, this is the correct way of accessing the static objects from a class

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java [hunk 2]
```diff
@@ -784,15 +785,16 @@ public CohortDefinition getMOHPatientsWithVLRequestorResultBetweenClinicalConsul
             + "         p.voided = 0 AND e.voided = 0  "
             + "             AND e.encounter_type = ${6}  "
             + "             AND e.location_id = :location  "
-            + "             AND e.encounter_datetime BETWEEN :startDate AND :endDate  "
+            + "             AND e.encounter_datetime > :startDate AND e.encounter_datetime <= :endDate  "
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java:784]: So the lower date bound is NOT considered here?

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java [hunk 3]
```diff
@@ -504,7 +504,7 @@ public CohortDefinition getMOHPatientsOnTreatmentFor6Months(
       query +=
           "             AND o.value_coded IS NOT NULL  "
               + "             AND DATE(o.obs_datetime) <= DATE(clinical.last_visit)  "
-              + "         AND DATEDIFF(clinical.last_visit, o.obs_datetime) >= 180  "; // check
+              + "         AND DATE(o.obs_datetime) <= DATE_SUB(clinical.last_visit,INTERVAL 6 MONTH) "; // check
```
Comment (ningosi) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java]: Not sure if 6 months is equivalent to 180 days, 6 calendar months are different from the days, please confirm

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java (hunk 2 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java:784]: Yes, according to the specs the lower date bound should not be considered

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java (hunk 1 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/QualityImprovement2020CohortQueries.java:2145]: Great

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java (hunk 3 above)
Comment (iteliolucas) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java]: Yes there is a slight difference for SQL, hence the result using days to calculate was not matching FGH's results in this particular indicator, using SQL MONTH instead, matches the results.

Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CommonCohortQueries.java]: Mhhh, that is fine then, please ressolve the confilcts, then I will merge this

**Review by ningosi — APPROVED** [review]

**Pull request merged**